src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3979,11 +3979,13 @@ public static bool ConnectAsync(SocketType socketType, ProtocolType protocolType
             }
             if (e.HasMultipleBuffers)
             {
-                throw new ArgumentException(SR.net_multibuffernotsupported, "BufferList");
+                throw new ArgumentException(SR.net_multibuffernotsupported, nameof(e));
             }
             if (e.RemoteEndPoint == null)
             {
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly, the argument not applicable
                 throw new ArgumentNullException("remoteEP");
+#pragma warning restore CA2208
```

Comment:
Maybe I missed it described somewhere.  Is your plan with this PR to temporarily disable/restore around cases we're not sure about and then subsequently audit all of them?

Reply:
Yest, I have suppressed:
1. The exceptions having any comments about compatability
2. The cases i am not sure about, including the one related to `ArgumentNullException(nameof(parameter.Property))` case

Reply:
Update: 
for 1: updated exceptions regardless compatibility
for 2: updated to use ArgumentException, some of them might be breaking change including this one. Doc needs to be updated for this one